src/promptflow-devkit/promptflow/_proxy/_python_inspector_proxy.py

```diff
@@ -36,6 +37,9 @@ def get_entry_meta(
     ) -> Dict[str, Any]:
         timeout = kwargs.get("timeout", FLOW_META_JSON_GEN_TIMEOUT)
         load_in_subprocess = kwargs.get("load_in_subprocess", True)
+        # if it's from eval batch run, use the same process to generate flow meta for sake of performance
+        if is_eval_batch_run():
+            load_in_subprocess = False
```

Comment:
The aim to open new process is to avoid influencing `sys.path` and `working dir` in other threads.
We will change `working dir` to python file's folder and append it to `sys.path`.

Although `load_in_subprocess` could be set from kwargs, seems like we don't have usage to set it as false for now.
https://github.com/search?q=repo%3Amicrosoft%2Fpromptflow%20get_entry_meta&type=code

So, maybe you need to validate if it's safe to load in current process in your situation, maybe it has issue for some corner case.

Reply:
By the way, seems like this change is not related to other parts of this PR. Maybe separate it as a new PR is better. (handy for revert at least)

Reply:
Thanks for the info. I also heard from @D-W- earlier that the VS Code extension had this issue before, thus load_in_process is being used. For evals, so far, I haven't run into any issues from my tests. I think the chance of module conflicts from evaluators probably is lower. But, sounds good, let me raise a separate PR for this change.